src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -249,81 +315,507 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
+        ///
+        /// <paramref name="second"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="millisecond"/> is less than 0 or greater than 999.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="microsecond"/> is less than 0 or greater than 900.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="kind"/> is not one of the <see cref="DateTimeKind"/> values.
+        /// </exception>
+        /// <remarks>
+        /// The allowable values for <paramref name="year"/>, <paramref name="month"/>, and <paramref name="day"/> parameters
+        /// depend on the <paramref name="calendar"/> parameter. An exception is thrown if the specified date and time cannot
+        /// be expressed using <paramref name="calendar"/>.
+        ///
+        /// For applications in which portability of date and time data or a limited degree of time zone awareness is important,
+        /// you can use the corresponding <see cref="DateTimeOffset"/> constructor.
+        /// </remarks>
+        public DateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int microsecond, Calendar calendar!!, DateTimeKind kind)
         {
             if ((uint)millisecond >= MillisPerSecond) ThrowMillisecondOutOfRange();
             if ((uint)kind > (uint)DateTimeKind.Local) ThrowInvalidKind();
 
             if (second != 60 || !s_systemSupportsLeapSeconds)
             {
-                ulong ticks = DateToTicks(year, month, day) + TimeToTicks(hour, minute, second);
-                ticks += (uint)millisecond * (uint)TicksPerMillisecond;
-                Debug.Assert(ticks <= MaxTicks, "Input parameters validated already");
+                var dateTime = calendar.ToDateTime(year, month, day, hour, minute, second, millisecond);
+                dateTime = dateTime.AddMicroseconds(microsecond);
+                ulong ticks = dateTime.UTicks;
                 _dateData = ticks | ((ulong)kind << KindShift);
             }
             else
             {
                 // if we have a leap second, then we adjust it to 59 so that DateTime will consider it the last in the specified minute.
-                this = new DateTime(year, month, day, hour, minute, 59, millisecond, kind);
+                this = new DateTime(year, month, day, hour, minute, 59, millisecond, microsecond, calendar, kind);
                 ValidateLeapSecond();
             }
         }
 
-        // Constructs a DateTime from a given year, month, day, hour,
-        // minute, and second for the specified calendar.
-        //
-        public DateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, Calendar calendar!!)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ulong Init(int year, int month, int day, int hour, int minute, int second, int millisecond)
```

Comment:
>private static ulong Init [](http://example.com/codeflow?start=8&length=25)

why we need this method? wouldn't 

private static ulong Init(int year, int month, int day, int hour, int minute, int second, int millisecond, DateTimeKind kind)

is enough to use. can specify the DateTimeKind.Unspecified. 

Actually you can default the last parameter of Init 
DateTimeKind kind = DateTimeKind.Unspecified?